Updates to RTD documentation based on full review

KateFriedman-NOAA opened the pull request:
**Description**

This PR updates the new RTD documentation that was moved from the global-workflow wiki. The wiki (and now RTD) needed to be reviewed for updates related to recent GFS/global-workflow system updates, as well as fix some spelling/syntax issues. This is the first documentation update since the new RTD documentation was added to the repo.

As part of the updates the following changes occurred:
- A few spelling and/or Sphinx syntax corrections.
- Introduce use of `note` and `warning` boxes.
- Review and update of content to match GFS system updates.
- Reorganize/rename some sections.

Preview of incoming updated documentation can be viewed here on RZDM:

https://www.emc.ncep.noaa.gov/gmb/emc.glopara/read-the-docs/global-workflow/issue1272/index.html

Note: Pages will look a bit different (e.g. missing bullets), likely due to no Sphinx installation on RZDM (?). Expect other similar subtle differences compared to [live RTD documentation page](https://global-workflow.readthedocs.io/en/latest/). I'm open to suggestions to make this preview look more like the live pages (e.g. install Sphinx?).

**Type of change**

Maintenance and regular updates to documentation.

**How Has This Been Tested?**

Viewed feature branch Git Action artifact generated html pages in browser.
  
Resolves #1272

KateFriedman-NOAA committed:
Correct and update documentation

- Reviewed documentation to correct spelling mistakes
and update contents for current state of system.
- Introduced a couple tables where needed.
- More updates to come.

Refs #1272

KateFriedman-NOAA committed:
Add note blocks

- Replace "Note" comments with ".. note::" blocks.

Refs #1272

KateFriedman-NOAA committed:
Convert tables in setup.rst

- Change table syntax to sphinx syntax.
- Remove extra line in index.rst.

Refs #1272

KateFriedman-NOAA committed:
Update table syntax in setup.rst

Refs #1272

KateFriedman-NOAA committed:
Update setup table syntax

Refs #1272

KateFriedman-NOAA committed:
Adjust setup table syntax again

Refs #1272

KateFriedman-NOAA committed:
Adjust setup table syntax yet again

Refs #1272

KateFriedman-NOAA committed:
Adjust setup table syntax once again

Refs #1272

KateFriedman-NOAA committed:
Trying different syntax for setup table

Refs #1272

KateFriedman-NOAA committed:
Adjust setup table syntax yet again again

Refs #1272

KateFriedman-NOAA committed:
Adjust setup table syntax once again again

Refs #1272

KateFriedman-NOAA committed:
Adjust setup table syntax

Refs #1272

KateFriedman-NOAA committed:
Several RTD documentation updates

- Update to use header row in tables.
- Update configuration table contents.
- Update example output from setup commands.
- Update warning text for crontab command.

Refs #1272

KateFriedman-NOAA committed:
Update setup script examples

Refs #1272

KateFriedman-NOAA committed:
Few updates to RTD documentation

- Update setup script name in configure.rst.
- Update python missing error text in errors_faq.rst.

Refs #1272

KateFriedman-NOAA committed:
Add development tools section

Refs #1272

KateFriedman-NOAA committed:
Fix case in new development section

Refs #1272

KateFriedman-NOAA committed:
Fix syntax in new development section

Refs #1272

KateFriedman-NOAA committed:
Retrying list table format with code in setup

Refs #1272

KateFriedman-NOAA committed:
Update code block in setup table

Refs #1272

KateFriedman-NOAA committed:
Update setup tables to list format

Refs #1272

KateFriedman-NOAA committed:
Correct rocoto table in setup

Refs #1272

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/update-rtd

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/update-rtd

KateFriedman-NOAA committed:
Update initial condition documentation

Refs #1272

KateFriedman-NOAA committed:
Small updates to init document

Refs #1272

KateFriedman-NOAA committed:
Small corrections to init document

Refs #1272

KateFriedman-NOAA committed:
Update table header in components

Refs #1272

WalterKolczynski-NOAA commented on docs/source/clone.rst:
```diff
@@ -32,31 +39,45 @@ Clone workflow and component repositories
    git clone https://github.com/NOAA-EMC/global-workflow.git
 
-ssh method (using a password protected SSH key)::
+ssh method (using a password protected SSH key):
```
Let's move this ahead of https and note it is the highly preferred method

KateFriedman-NOAA replied:
Done @ 75f19424

WalterKolczynski-NOAA commented on docs/source/configure.rst:
```diff
@@ -2,70 +2,61 @@
 +----------------+------------------------------+---------------+-------------+---------------------------------------------------+
-| WAFSF          | Run jobs to produce WAFS     | NO            | YES         | downstream processing, ops only                   |
-|                | products                     |               |             |                                                   |
+| DO_WAVE        | Runs wave jobs and produce   | YES           | YES         | One-way wave coupling, new for GFSv16             |
```
This is control by APP and probably doesn't need to be on this list.

KateFriedman-NOAA replied:
Good point. Committed to branch at 3f9b7012.

WalterKolczynski-NOAA commented on docs/source/monitor_rocoto.rst:
```diff
@@ -53,9 +70,10 @@ Usage
+   Note 1: Terminal/window must be wide enough to display all experiment information columns, viewer will complain if not.
 
-Note 2: The viewer requires the full path to the database and xml files if you are not in your EXPDIR when you invoke it.
+   Note 2: The viewer requires the full path to the database and xml files if you are not in your EXPDIR when you invoke it.
```
Note 3: Only `TERM=xterm` is supported. You may wish to create a shell function to switch automatically if you are in a different terminal:

Bash example
```
function rv {
	oldterm=${TERM};
	export TERM='xterm';
	${PATH_TO_VIEWER}/rocoto_viewer.py $@;
	export TERM=${oldterm};
}
```

KateFriedman-NOAA replied:
Added @ 6be21829

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/update-rtd

KateFriedman-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/update-rtd

KateFriedman-NOAA committed:
Update clone method section in clone.rst

- Move ssh method above https and mention that ssh method
is highly preferred.

Refs #1272

KateFriedman-NOAA committed:
Remove DO_WAVE from configure table

- Since DO_WAVE is controlled by the APP switch don't need
to mention DO_WAVE in the switch configuration table.

Refs #1272

KateFriedman-NOAA committed:
Add note 3 to rocoto viewer section

- Note 3 states that TERM=xterm is the only supported TERM.
- Include bash example of shell function to automatically
switch if user is in different terminal.

Refs #1272

KateFriedman-NOAA commented:
Preview of RTD updated after addressing @WalterKolczynski-NOAA's comments: https://www.emc.ncep.noaa.gov/gmb/emc.glopara/read-the-docs/global-workflow/issue1272/index.html
